cmd/ci-secret-bootstrap/main.go

```diff
@@ -858,38 +866,43 @@ func main() {
 		logrus.WithError(err).Fatal("Invalid arguments.")
 	}
 	if err := o.completeOptions(&censor); err != nil {
-		logrus.WithError(err).Fatal("Failed to complete options.")
+		logrus.WithError(err).Error("Failed to complete options.")
 	}
 	client, err := o.secrets.NewReadOnlyClient(&censor)
 	if err != nil {
 		logrus.WithError(err).Fatal("Failed to create client.")
 	}
-	logrus.RegisterExitHandler(func() {
+
+	if errs := reconcileSecrets(o, client); len(errs) > 0 {
+		logrus.WithError(utilerrors.NewAggregate(errs)).Fatalf("errors while updating secrets")
+	}
+}
+
+func reconcileSecrets(o options, client secrets.ReadOnlyClient) (errs []error) {
```

Comment:
Putting `if o.validateOnly {}` inside `reconcileSecrets` sounds a bit odd.
Maybe moving it out?

Reply:
It uses the client, that's why. I just moved all the code inside the new `reconcileSecrets`. If you have an idea for a better name, I am all ears. But we can not move it out because we need this function to define the scope of the `defer Logout` call.

Reply:
> you have an idea for a better name

Oh No. That is really not my strength.

/lgtm